src/Ng/Jobs/BlobSnapshotJob.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage; 
+using Microsoft.WindowsAzure.Storage.Blob;
+using NuGet.Services.Configuration;
+
+namespace Ng.Jobs
+{
+    /// <summary>
+    /// Offline job that can be used to take one snapshot for blobs in a container.
+    /// </summary>
+    public class BlobSnapshotJob : NgJob
+    {
+        private string _connectionString;
+        private string _container;
+        private bool _verbose;
+
+        public BlobSnapshotJob(ILoggerFactory loggerFactory) : base(loggerFactory)
+        {
+
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng blobsnapshot "
+                   + $"-{Arguments.ConnectionString} <connectionString> "
+                   + $"-{Arguments.Container} <container> "
+                   + $"[-{Arguments.Verbose} true|false]";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            _connectionString = arguments.GetOrThrow<string>(Arguments.ConnectionString);
+            _container = arguments.GetOrThrow<string>(Arguments.Container);
+            _verbose = arguments.GetOrThrow<bool>(Arguments.Verbose);
+        }
+        
+        protected override Task RunInternal(CancellationToken cancellationToken)
+        {
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(_connectionString);
+            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
+            TakeOneSnapshot(_container, blobClient);
+            return Task.FromResult(true);
+        }
+
+        private void TakeOneSnapshot(string containerName, CloudBlobClient client)
+        {
+            CloudBlobContainer container = client.GetContainerReference(containerName);
+            var blobList = container.ListBlobs(prefix:null, useFlatBlobListing:true, blobListingDetails:BlobListingDetails.None);
```

Comment:
This should use `ListBlobsSegmentedAsync` so the whole blob list is never in memory.